README.md

```diff
@@ -48,298 +48,400 @@ The FWS Payments contract enables ERC20 token payment flows through "rails" - au
+  - For the native token, `msg.value` must equal `amount`.
 
 #### `depositWithPermit(address token, address to, uint256 amount, uint256 deadline, uint8 v, bytes32 r, bytes32 s)`
 
-Deposits tokens using EIP-2612 permit.
+Deposits tokens using an EIP-2612 permit, allowing for gasless token approval.
 
+- **When to use:** Ideal for user-facing applications where the user can sign a permit off-chain. This combines approval and deposit into a single on-chain transaction, saving gas and improving user experience.
+- **Note:** This function is for ERC20 tokens only and does not support the native token.
 - **Parameters**:
-  - `token`: ERC20 token contract address supporting EIP-2612 permits
-  - `to`: Recipient account address (must be the signer of the permit)
-  - `amount`: Token amount to deposit
-  - `deadline`: Permit expiration timestamp
-  - `v`, `r`, `s`: Signature components for EIP-2612 permit signature
-
+  - `token`: ERC20 token contract address supporting EIP-2612 permits.
+  - `to`: The account address to credit (must be the signer of the permit).
+  - `amount`: Token amount to deposit.
+  - `deadline`: Permit expiration timestamp.
+  - `v`, `r`, `s`: Signature components for the EIP-2612 permit.
 - **Requirements**:
-  - Token must support EIP-2612 permit
-  - Caller must have signed the permit
-  - Permit must be valid and not expired
+  - Token must support EIP-2612.
+  - `to` must be `msg.sender` (the one submitting the transaction).
 
-#### `depositWithPermitAndApproveOperator(address token, address to, uint256 amount, uint256 deadline, uint8 v, bytes32 r, bytes32 s, address operator, uint256 rateAllowance, uint256 lockupAllowance, uint256 maxLockupPeriod)`
+#### `depositWithPermitAndApproveOperator(...)`
 
-Deposits tokens using EIP-2612 permit and sets operator approval in a single transaction.
+A powerful convenience function that combines three actions into one transaction:
+1. Approves token spending via an EIP-2612 permit.
+2. Deposits tokens into the specified account.
+3. Sets approval for an operator.
 
-- **Parameters**:
-  - `token`: ERC20 token contract address supporting EIP-2612 permits
-  - `to`: Recipient account address (must be the signer of the permit)
-  - `amount`: Token amount to deposit
-  - `deadline`: Permit expiration timestamp
-  - `v`, `r`, `s`: Signature components for EIP-2612 permit signature
-  - `operator`: Address to grant permissions to
-  - `rateAllowance`: Maximum payment rate the operator can set across all rails
-  - `lockupAllowance`: Maximum funds the operator can lock for future payments
-  - `maxLockupPeriod`: Maximum allowed lockup period in epochs
+- **When to use:** This is the most efficient way for a new user to get started. It funds their account and authorizes a service contract (operator) in a single step.
+- **Note:** This function is for ERC20 tokens only.
+- *(Parameters are as documented in the contract interface)*
```

Comment:
disagree, I'd really like these to be in the README like they were; I buy not putting them in the heading (tho I still would), but at least list them